docs/user/OIDC-publishing/security-model.md

```diff
@@ -0,0 +1,113 @@
+---
+title: Security model and considerations
+---
+
+{{ preview('oidc-publishing') }}
+
+# Security model and considerations
+
+## Security model
+
+GitHub Actions' own security model for OpenID Connect tokens is a little subtle:
+
+* Any workflow defined in a repository can request an OIDC token,
+ *with any audience*, **so long as it has the `id-token: write` permission**.
+
+* The claims defined in an OIDC token are *bound to the workflow*, meaning
+  that a workflow defined at `foo.yml` in `org/repo` **cannot impersonate**
+  a workflow defined at `bar.yml` in `org/repo`. However, if `foo.yml` is
+  *renamed* to `bar.yml`, then the *new* `bar.yml` will be indistinguishable
+  from the old `bar.yml` **except** for claims that reflect the repository's
+  state (e.g. `git` ref, branch, etc.).
+
+* *Generally speaking*, "third party" events **cannot** request an OIDC
+  token: even if they can trigger the workflow that requests the token,
+  the actual token retrieval step will fail. For example: PRs issued from forks
+  of a repository **cannot** access the OIDC tokens in the "upstream"
+  repository's workflows.
+
+  * The exception to this is `pull_request_target` events, which are
+    **[fundamentally dangerous] by design** and should not be used without
+    careful consideration.
+
+## Considerations
+
+While more secure than passwords and long-lived API tokens, OIDC publishing
+is not a panacea. In particular:
+
+* Short-lived API tokens are still sensitive material, and should not be
+  disclosed (ideally not at all, but certainly not before they expire).
+
+* OIDC tokens themselves are sensitive material, and should not be disclosed.
+  OIDC tokens are also short-lived, but an attacker who successfully intercepts
+  one can mint API tokens against it for as long as it lives.
+
+* Configuring an OIDC publisher means establishing trust in a particular piece
+  of external state; that state **must not** be controllable by untrusted
+  parties. In particular, for OIDC publishing with GitHub Actions, you **must**:
+
+  * Trust the correct username and repository: if you trust a repository
+    other than one you control and trust, that repository can upload to your
+    PyPI project.
+
+  * Trust the correct workflow: you shouldn't trust every workflow
+    to upload to PyPI; instead, you should isolate responsibility to the
+    smallest (and least-privileged) possible separate workflow. We recommend
+    naming this workflow `release.yml`.
+
+  * Take care when merging third-party changes to your code: if you trust
+    `release.yml`, then you must make sure that third-party changes to that
+    workflow (or code that runs within that workflow) are not malicious.
```

Comment:
Probably needs a bit more indentation:

<img width="699" alt="image" src="https://user-images.githubusercontent.com/1324225/227343994-99fae74f-1343-437c-b92f-d3c4192b9b3a.png">


```suggestion
   * Trust the correct username and repository: if you trust a repository
     other than one you control and trust, that repository can upload to your
     PyPI project.

   * Trust the correct workflow: you shouldn't trust every workflow
     to upload to PyPI; instead, you should isolate responsibility to the
     smallest (and least-privileged) possible separate workflow. We recommend
     naming this workflow `release.yml`.

   * Take care when merging third-party changes to your code: if you trust
     `release.yml`, then you must make sure that third-party changes to that
     workflow (or code that runs within that workflow) are not malicious.
```

Reply:
Hmm, this indentation now looks okay in the GitHub preview (https://github.com/trail-of-forks/warehouse/blob/tob-oidc-user-docs/docs/user/OIDC-publishing/security-model.md), but MyST needs some more mangling...

https://docspypiorg--13285.org.readthedocs.build/OIDC-publishing/security-model/